Libhb vce h265 10bit

OvchinnikovDmitrii opened the pull request:
**Description of Change:**

Added H.265 10 bit(AMD VCE) encoder. issue - https://github.com/HandBrake/HandBrake/issues/4537



**Test on:**

- [x] Windows 10+  (via MinGW)
- [ ] macOS 10.13+
- [ ] Ubuntu Linux

OvchinnikovDmitrii committed:
contrib/ffmpeg: 10 bit encode support for vce h265.

sr55 commented:
Hi, Thanks for the patch.

Have the ffmpeg patches been submitted upstream yet?

I'll review over & test over the weekend if I get a moment.

galad87 commented on libhb/handbrake/common.h:
```diff
@@ -563,6 +563,7 @@ struct hb_job_s
 #define HB_VCODEC_FFMPEG_VP9   0x0000080
 #define HB_VCODEC_FFMPEG_VCE_H264 0x00040000
 #define HB_VCODEC_FFMPEG_VCE_H265 0x00080000
+#define HB_VCODEC_FFMPEG_VCE_H265_10BIT 0x00080001
```
HB_VCODEC_ is kind of a bit field, and each codec should be a single bit. But I'll have to reassign them anyway for the VP9 10 bit pull request, so let's wait a bit to fix this.

galad87 commented on libhb/common.c:
```diff
@@ -283,6 +283,7 @@ hb_encoder_internal_t hb_video_encoders[]  =
     { { "H.265 (Intel QSV)",           "qsv_h265",         "H.265 (Intel Media SDK)",        HB_VCODEC_QSV_H265,                           HB_MUX_MASK_MP4|HB_MUX_MASK_MKV, }, NULL, 0, 1, HB_GID_VCODEC_H265_QSV,   },
     { { "H.265 10-bit (Intel QSV)",    "qsv_h265_10bit",   "H.265 10-bit (Intel Media SDK)", HB_VCODEC_QSV_H265_10BIT,                     HB_MUX_MASK_MP4|HB_MUX_MASK_MKV, }, NULL, 0, 1, HB_GID_VCODEC_H265_QSV,   },
     { { "H.265 (AMD VCE)",             "vce_h265",         "H.265 (AMD VCE)",                HB_VCODEC_FFMPEG_VCE_H265,                    HB_MUX_MASK_MP4|HB_MUX_MASK_MKV, }, NULL, 0, 1, HB_GID_VCODEC_H265_VCE,   },
+    { { "H.265 10-bit(AMD VCE)",       "vce_h265_10bit",   "H.265 (AMD VCE)",                HB_VCODEC_FFMPEG_VCE_H265_10BIT,              HB_MUX_MASK_MP4|HB_MUX_MASK_MKV, }, NULL, 0, 1, HB_GID_VCODEC_H265_VCE,   },
```
It's missing a space before (AMD VCE)

sr55 replied:
The 3rd argument also needs to be "H.265 10-bit (AMD VCE)" otherwise it displays incorrectly in the log

OvchinnikovDmitrii commented:
> Have the ffmpeg patches been submitted upstream yet?

I have  send these patches to ffmpeg-devel(https://patchwork.ffmpeg.org/project/ffmpeg/list/?series=7411) 
but there was no reaction at the moment.

sr55 commented:
Validated on Windows. Seems to work quite nicely.  If the cosmetic issue mention above is fixed, I'm happy for this to go in.

sr55 commented:
Looks like we may need a quick resolve on this PR as well please.

OvchinnikovDmitrii committed:
Cosmetic fixes.

OvchinnikovDmitrii commented:
> Validated on Windows. Seems to work quite nicely. If the cosmetic issue mention above is fixed, I'm happy for this to go in.

Thank you for the positive feedback! I corrected the cosmetic issues in the encoder registration.  What to do with "HB_VCODEC_FFMPEG_VCE_H265_10 BIT 0x00080001" ? have the values already been reassigned?( for the VP9 10 beat pull request)

galad87 commented:
They will in #4578 , which should be merged soon after a bit of testing.

galad87 commented:
#4578 has been merged, if you can rebase it and move the pixel format entry to encavcodec.c it should be good to go.

OvchinnikovDmitrii committed:
Merge branch 'master' into libhb_vce_h265_10bit

# Conflicts:
#	libhb/common.c
#	libhb/encavcodec.c
#	libhb/handbrake/common.h

OvchinnikovDmitrii committed:
libhb: build fix and cleanup.

OvchinnikovDmitrii commented:
> #4578 has been merged, if you can rebase it and move the pixel format entry to encavcodec.c it should be good to go.

Thanks! I rebased and moved the definition vce_pix_formats_10bit.

galad87 commented:
Tested on Windows and merged, thanks!
